apps/backoffice-v2/src/pages/TransactionMonitoringAlerts/components/AlertsAssignDropdown/AlertsAssignDropdown.tsx

```diff
@@ -0,0 +1,84 @@
+import React, { FunctionComponent, useMemo } from 'react';
+import { TUsers } from '@/domains/users/types';
+import { DoubleCaretSvg, UnassignedAvatarSvg } from '@/common/components/atoms/icons';
+import { UserAvatar } from '@/common/components/atoms/UserAvatar/UserAvatar';
+import { Dropdown } from '@/common/components/molecules/Dropdown/Dropdown';
+
+export const AlertsAssignDropdown: FunctionComponent<{
+  assignees: TUsers;
+  authenticatedUserId: string;
+  isDisabled: boolean;
+  isAssignedToMe: boolean;
+  onAssigneeSelect: (id: string | null, isAssignedToMe: boolean) => () => void;
+}> = ({ assignees, authenticatedUserId, isDisabled, onAssigneeSelect }) => {
+  const sortedAssignees = useMemo(
+    () =>
+      // Sort assignees so that the authenticated user is always first
+      assignees
+        ?.slice()
+        ?.sort((a, b) =>
+          a?.id === authenticatedUserId ? -1 : b?.id === authenticatedUserId ? 1 : 0,
+        ),
+    [assignees, authenticatedUserId],
+  );
+
+  const options = useMemo(() => {
+    return [
+      {
+        id: 'unassign',
+        value: null,
+      },
+      ...(sortedAssignees?.map(assignee => ({
+        id: assignee?.id,
+        value: assignee,
+      })) ?? []),
+    ];
+  }, [sortedAssignees]);
```

Comment:
Im a bit confusing why our component cant work well with the original assignees list?

im not familiar with a convention of having object in a value property.

Usually when an event is occurred by the component like onClick/OnChange then the item value will be present on `e.target.value`